extensions/fbc/src/org/sbml/jsbml/ext/fbc/converters/FbcV1ToCobraConverter.java

```diff
@@ -65,69 +70,87 @@ public SBMLDocument convert(SBMLDocument sbmlDocument) throws SBMLException {
-        if (species.isSetPlugin("fbc")) {
-          FBCSpeciesPlugin fbcSpeciesPlugin = (FBCSpeciesPlugin)species.getPlugin("fbc");
+        if (species.isSetPlugin(FBCConstants.shortLabel)) {
+          FBCSpeciesPlugin fbcSpeciesPlugin = (FBCSpeciesPlugin)species.getPlugin(FBCConstants.shortLabel);
           
           if (fbcSpeciesPlugin.isSetChemicalFormula()) {
             String chemicalFormula = fbcSpeciesPlugin.getChemicalFormula();
             Properties pElementsSpeciesNotes = new Properties();
             pElementsSpeciesNotes = CobraUtil.parseCobraNotes(species);
-            if (pElementsSpeciesNotes.getProperty("FORMULA") != null) {
-              pElementsSpeciesNotes.remove("FORMULA");
+            if (pElementsSpeciesNotes.getProperty(CobraConstants.FORMULA) != null) {
+              pElementsSpeciesNotes.remove(CobraConstants.FORMULA);
             }
-            pElementsSpeciesNotes.setProperty("FORMULA", chemicalFormula);
+            pElementsSpeciesNotes.setProperty(CobraConstants.FORMULA, chemicalFormula);
             CobraUtil.writeCobraNotes(species, pElementsSpeciesNotes);
           }
         
           if (fbcSpeciesPlugin.isSetCharge()) {
             int charge = fbcSpeciesPlugin.getCharge();
             Properties pElementsSpeciesNotes = new Properties();
             pElementsSpeciesNotes = CobraUtil.parseCobraNotes(species);
-            if (pElementsSpeciesNotes.getProperty("CHARGE") != null) {
-              pElementsSpeciesNotes.remove("CHARGE");
+            if (pElementsSpeciesNotes.getProperty(CobraConstants.CHARGE) != null) {
+              pElementsSpeciesNotes.remove(CobraConstants.CHARGE);
             }
-            pElementsSpeciesNotes.setProperty("CHARGE", Integer.toString(charge));
+            pElementsSpeciesNotes.setProperty(CobraConstants.CHARGE, Integer.toString(charge));
             CobraUtil.writeCobraNotes(species, pElementsSpeciesNotes);
           }
-          species.unsetPlugin("fbc");
+          species.unsetPlugin(FBCConstants.shortLabel);
         }
       }
       
-      // get lower and upper flux bound from the list of flux bounds and write them in the kinetic law; delete the list of flux bounds
-      FBCModelPlugin fbcModelPlugin = (FBCModelPlugin)model.getPlugin("fbc");
-      Set<String> listOFBReactionIds = new HashSet<String>();
+      // Process reactions
+      // 1. Collect lower and upper flux bounds from the list of flux bounds
+      FBCModelPlugin fbcModelPlugin = (FBCModelPlugin)model.getPlugin(FBCConstants.shortLabel);
+      Map<String, Double> reactionIdToLowerFluxBoundDoubleValueMap = new HashMap<>();
+      Map<String, Double> reactionIdToUpperFluxBoundDoubleValueMap = new HashMap<>();
+      
       for (FluxBound fluxBound : fbcModelPlugin.getListOfFluxBounds()) {
-        listOFBReactionIds.add(fluxBound.getReaction());
+        if (fluxBound.getOperation().equals(FluxBound.Operation.GREATER_EQUAL))
```

Comment:
I think you should consider other operators as well, check the specification to be able to cover all the cases.

Reply:
I now added all possible operations according to the specification and added missing ones to the FluxBound.Operations enumeration.